src/Controls/tests/Core.UnitTests/Gestures/SwipeGestureRecognizerTests.cs

```diff
@@ -117,5 +117,45 @@ public void SwipedEventDirectionMatchesTotalYTestWithFlags()
 			((ISwipeGestureController)swipe).DetectSwipe(view, SwipeDirection.Up | SwipeDirection.Down);
 			Assert.Equal(SwipeDirection.Up, direction);
 		}
+
+		[Theory]
+		[InlineData(100.0, 0.0, 90.0, 0.0, 100.0)]
+		[InlineData(100.0, 100.0, 90.0, -100.0, 100.0)]
+		[InlineData(100.0, 0.0, 180.0, -100.0, 0.0)]
+		[InlineData(100.0, 100.0, 270.0, 100.0, -100.0)]
+		public void TransformSwipeCoordinatesWithRotation_ReturnsTransformedCoordinates(
+			double inputX, double inputY, double rotation, double expectedX, double expectedY)
+		{
+			var result = Internals.SwipeGestureExtensions.TransformSwipeCoordinatesWithRotation(inputX, inputY, rotation);
+
+			Assert.Equal(expectedX, result.x, 1);
+			Assert.Equal(expectedY, result.y, 1);
+		}
+
+		[Theory]
+		[InlineData(SwipeDirection.Up, 90.0, SwipeDirection.Right)]
+		[InlineData(SwipeDirection.Right, 90.0, SwipeDirection.Down)]
+		[InlineData(SwipeDirection.Down, 90.0, SwipeDirection.Left)]
+		[InlineData(SwipeDirection.Left, 90.0, SwipeDirection.Up)]
+		public void TransformSwipeDirectionForRotation_90DegreeClockwise_ReturnsCorrectDirection(
+			SwipeDirection direction, double rotation, SwipeDirection expected)
+		{
+			var result = Internals.SwipeGestureExtensions.TransformSwipeDirectionForRotation(direction, rotation);
+
+			Assert.Equal(expected, result);
+		}
+
+		[Theory]
+		[InlineData(SwipeDirection.Up, 180.0, SwipeDirection.Down)]
+		[InlineData(SwipeDirection.Right, 180.0, SwipeDirection.Left)]
+		[InlineData(SwipeDirection.Down, 180.0, SwipeDirection.Up)]
+		[InlineData(SwipeDirection.Left, 180.0, SwipeDirection.Right)]
+		public void TransformSwipeDirectionForRotation_180Degree_ReturnsCorrectDirection(
```

Comment:
Include edge cases like

```
[InlineData(SwipeDirection.Up, double.NaN, SwipeDirection.Up)]
[InlineData(SwipeDirection.Right, double.PositiveInfinity, SwipeDirection.Right)]
[InlineData(SwipeDirection.Down, double.NegativeInfinity, SwipeDirection.Down)]
```

Reply:
I have added the testcases for infinity and NaN value based on your suggestion.